### PR TITLE
[video][android] display subtitles button when subtitles exist

### LINKED ISSUE #31802: [expo-video] Subtitle settings in player are not available on Android
### Minimal reproducible example

https://github.com/ConnerPC88/expo-video-min

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a development build

#### Expected Behavior:

    Start an .mp4 video playback that contains subtitle data.
    Tap in the player to bring up controls.
    Be able to find and select subtitles (enable/disable/language).

#### Current Behavior:

    Start an .mp4 video playback that contains subtitle data.
    Tap in the player to bring up controls.
    No subtitle options are shown on Android (but are on iOS)

#### Extra info

As an example m3u8 file with captions, I'm using: [this file](https://devstreaming-cdn.apple.com/videos/streaming/examples/img_bipbop_adv_example_fmp4/master.m3u8)

If you use [this site](https://castr.com/hlsplayer/) to test the above file, you can select subtitle languages or turn them off. 

iOS **_DOES_** have subtitle controls (off/on/lang switch), but Android does not have any controls for those options.

### Environment

```text
System:
  OS: Windows 11 10.0.22631
  CPU: (16) x64 12th Gen Intel(R) Core(TM) i7-1260P
  Memory: 1.84 GB / 15.70 GB
Binaries:
  Node:
    version: 20.15.0
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 4.3.1
    path: C:\Program Files\nodejs\yarn.CMD
  npm:
    version: 10.7.0
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
  Windows SDK: Not Found
IDEs:
  Android Studio: Not Found
  Visual Studio:
    - 17.10.34928.147 (Visual Studio Community 2022)
Languages:
  Java: 17.0.11
  Ruby:
    version: 3.3.2
    path: C:\Ruby33-x64\bin\ruby.EXE
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.2
    wanted: 0.74.2
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Expo Doctor Diagnostics

```text
This project contains native project folders but also has native configuration properties in app.json, indicating it is configured to use Prebuild. When the android/ios folders are present, if you don't run prebuild in your build pipeline, the following properties will not be synced: plugins, android.
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for issues with metro config
✔ Check if the project meets version requirements for submission to app stores
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check npm/ yarn versions
✔ Check that packages match versions required by installed Expo SDK
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK
```

### PR DESCRIPTION
**Pull request by fobos531**
# Why

Fixes https://github.com/expo/expo/issues/31802

iOS has this built in so we're looking to achieve parity on Android. This PR focuses **solely** on giving the ability to select the CC in the UI. In future PRs this functionality can be made much better by allowing to fetch/set subtitle tracks from JS, allowing subtitle styling on Android etc., but this PR I would consider to be the bare minimum.

# How

The `playerView` has a `setShowSubtitleButton` method that is now controlled by `useNativeControls` prop.

# Test Plan

NCL with a video that has subtitles: https://devstreaming-cdn.apple.com/videos/streaming/examples/img_bipbop_adv_example_fmp4/master.m3u8

https://github.com/user-attachments/assets/a732aab1-8bae-4524-91cc-c33ebcedb73c

**Note** that the CC button is displayed even if the video has no subtitle tracks. I thought of maybe listening to player events for `onTracksChanged` and intelligently displaying the button ONLY if it has subtitle tracks, but I didn't want to get ahead of myself and kept it simple. I did try a few options but was unsuccessful. However, if you have ideas on how to do this properly, feel free to share them.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by behenate — CHANGES_REQUESTED** [review]
I can already feel all the people complaining of the constantly visible subtitles button 🤣 . I've spend some time figuring out how to display it contidionally. Take a look at these changes and apply them to your PR

https://github.com/expo/expo/compare/main...%40behenate/conditional-subtitles

**Comment by fobos531** [comment]
> I can already feel all the people complaining of the constantly visible subtitles button 🤣 . I've spend some time figuring out how to display it contidionally. Take a look at these changes and apply them to your PR
> 
> [main...@behenate/conditional-subtitles](https://github.com/expo/expo/compare/main...@behenate/conditional-subtitles)

Hello @behenate , yeah, makes perfect sense 😂

Your changes work perfectly, nice find!

**Comment by behenate** [comment]
Let's just get an approve from @lukmccall before merging 😄

**Review by behenate — CHANGES_REQUESTED** [review]

File: packages/expo-video/android/src/main/res/layout/fullscreen_player_activity.xml [hunk 1]
```diff
@@ -11,6 +11,5 @@
   <androidx.media3.ui.PlayerView
     android:id="@+id/player_view"
     android:layout_width="match_parent"
-    android:layout_height="match_parent" />
```
Comment (behenate) [inline packages/expo-video/android/src/main/res/layout/fullscreen_player_activity.xml:14]: You deleted a bit too much 😅

**Review by lukmccall — APPROVED** [review]
Looking good!
There is one more comment from @behenate; after applying it, this PR should be ready to be merged.

File: packages/expo-video/android/src/main/res/layout/fullscreen_player_activity.xml (hunk 1 above)
Comment (fobos531) [reply packages/expo-video/android/src/main/res/layout/fullscreen_player_activity.xml:14]: woops, sorry about that @behenate  🙈🙈🙈 fixed now

**Comment by fobos531** [comment]
Hey @lukmccall - done, should be good now!

File: packages/expo-video/android/src/main/res/layout/fullscreen_player_activity.xml [hunk 2]
```diff
@@ -13,4 +13,4 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent" />
 
-</FrameLayout>
+</FrameLayout>
```
Comment (lukmccall) [inline packages/expo-video/android/src/main/res/layout/fullscreen_player_activity.xml:16]:
```suggestion
</FrameLayout>

```

**Pull request merged**